### PR TITLE
added query entailment analyzer phase

### PR DESCRIPTION
**Pull request by patofp**
# Description

Added query entailment inference phase and join query and schema entailments phase.

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/inference/IntersectEntailmentAnalyzer.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.gsk.kg.engine.inference
+
+import com.gsk.kg.config.InferenceEntailment
+
+object IntersectEntailmentAnalyzer {
+
```
Comment (TomyMeren) [inline modules/engine/src/main/scala/com/gsk/kg/engine/inference/IntersectEntailmentAnalyzer.scala:6]: Is it necessary to create this object only for this sorting method?

**Review by TomyMeren — APPROVED** [review]
LGTM!

File: modules/engine/src/main/scala/com/gsk/kg/engine/inference/IntersectEntailmentAnalyzer.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/inference/IntersectEntailmentAnalyzer.scala:6]: Not at all! But as it is a Phase on it's own I decided to create it as an object instance as the rest of the Phases.
Captura de pantalla 2022-02-16 a las 16 49 19

Comment (TomyMeren) [reply modules/engine/src/main/scala/com/gsk/kg/engine/inference/IntersectEntailmentAnalyzer.scala:6]: Ok, It gives more clarity to the whole process 😄

**Pull request merged**